lib/Conversion/TorchToTosa/TosaLegalizeCommon.cpp

```diff
@@ -27,6 +28,370 @@
+  // dim2/last dim:	tensor([[   [ [0], [1] ],
+  //		                   	[ [0], [1] ],
+  //			            	[ [0], [1] ],
+  //		    	        	[ [0], [1] ], ]]) 1*4*2*1
+  auto indicesDim = getConstTensor<int32_t>(rewriter, op,
+                                            /*vec=*/indicesVec,
+                                            /*shape=*/indicesOneDimShape);
+  return indicesDim;
+}
+
+llvm::Optional<Value> convertTorchIndexToTfIndices(PatternRewriter &rewriter,
+                                                   Operation *op,
+                                                   Value paramsValue,
+                                                   Value indexValue,
+                                                   int32_t axis) {
+  // For easy understanding of this algorithm, the following comments are with
+  // an exact example: torch.aten.gather(!torch.vtensor<[1,4,3],f32>, axis=2,
+  // !torch.vtensor<[1,4,2],si64>) -> !torch.vtensor<[1,4,2],f32>
+  // https://gist.github.com/AmosLewis/2f18434397025211da4491735bcc6db6
+  //
+  // Convert Torch Index     to       TF Indices
+  //    [[                         [[   d0 d1  d2  d0 d1 d2
+  //        [0,0],                     [[0, 0, 0],[0, 0, 0]],
+  //        [1,0],                     [[0, 1, 1],[0, 1, 0]],
+  //        [2,1],                     [[0, 2, 2],[0, 2, 1]],
+  //        [2,1]                      [[0, 3, 2],[0, 3, 1]]
+  //    ]] 1*4*2                   ]] 1*4*2*3
+
+  auto paramsType = paramsValue.getType().dyn_cast<RankedTensorType>();
+  auto indexType = indexValue.getType().dyn_cast<RankedTensorType>();
+  auto paramsShape = paramsType.getShape(); // [1 4 3]
+  auto indexShape = indexType.getShape();   // [1 4 2]
+  int paramsRank = paramsShape.size();      // 3
+  int indexRank = indexShape.size();        // 3
+
+  // Initialize the final tf indices shape, and the shape of each dim that can
+  // concat to this tf indices
+  SmallVector<int64_t> indicesShape;       // [1 4 2 3]
+  SmallVector<int64_t> indicesOneDimShape; // [1 4 2 1]
+  for (auto shape : indexShape) {
+    indicesShape.push_back(shape);
+    indicesOneDimShape.push_back(shape);
+  }
+  indicesShape.push_back(paramsRank);
+  indicesOneDimShape.push_back(1);
+
+  // Get the chosen axis index
+  // indexValue reshape to indicesDim: shape append 1
+  // [1 4 2] -> [1 4 2 1]
+  // dim2:	tensor([[   [ [0], [0] ],
+  //			    [ [1], [0] ],
+  //			    [ [2], [1] ],
+  //			    [ [2], [1] ], ]]) 1*4*2*1
+  auto indicesChosenAxis = tosa::CreateOpAndInfer<tosa::ReshapeOp>(
+      rewriter, op->getLoc(),
+      GetTypeFromTensorShape(indicesOneDimShape, indexType.getElementType()),
+      indexValue, rewriter.getI64ArrayAttr(indicesOneDimShape));
+
+  SmallVector<Value> concatInputs;
+  for (auto dim = 0; dim < paramsRank; dim++) {
+    if (dim != axis) {
+      auto indices = createOneDimTfIndices(rewriter, op,
+                                           /*result shape*/ indicesOneDimShape,
```

Comment:
Comments like these should always match the argument names in the function definition. `createOneDimTfIndices` does not have a `result shape` named argument. In this case, no comment is needed since the variable being passed has the same name as the argument of the function

Reply:
Done